src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2038,6 +2042,12 @@ SubquerySqlTaskList(Job *job, PlannerRestrictionContext *plannerRestrictionConte
 	/* error if shards are not co-partitioned */
 	ErrorIfUnsupportedShardDistribution(subquery);
 
+	/* error if unsupported join on reference tables */
```

Comment:
Can we performs those checks before physical planner ?  Logical planner was the place we do those checks.